hal/network/lwip/wiznet/wiznetif_config.cpp

```diff
@@ -0,0 +1,163 @@
+    if (result != sizeof(tempData) ||
+            /* memcmp(tempData, wizNetifConfigData_, sizeof(tempData) != 0)) { // not reliable */
+            tempData.size != wizNetifConfigData_.size ||
+            tempData.cs_pin != wizNetifConfigData_.cs_pin ||
+            tempData.reset_pin != wizNetifConfigData_.reset_pin ||
+            tempData.int_pin != wizNetifConfigData_.int_pin ) {
+        // LOG(INFO, "Saving cached wizNetifConfigData, size: %d", wizNetifConfigData_.size);
+        // logWizNetifConfigData(wizNetifConfigData_);
+        result = SystemCache::instance().set(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&wizNetifConfigData_, sizeof(wizNetifConfigData_));
+    }
+    return (result < 0) ? result : SYSTEM_ERROR_NONE;
+}
+
+int WizNetifConfig::recallWizNetifConfigData() {
+    WizNetifConfigData tempData;
+    int result = SystemCache::instance().get(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&tempData, sizeof(tempData));
+    if (result != sizeof(tempData)) {
+        return SYSTEM_ERROR_NOT_FOUND;
+    }
+    // LOG(INFO, "Recalling cached wizNetifConfigData");
+    // logWizNetifConfigData(tempData);
+    memcpy(&wizNetifConfigData_, &tempData, sizeof(wizNetifConfigData_));
+    return SYSTEM_ERROR_NONE;
+}
+
+// int WizNetifConfig::deleteWizNetifConfigData() {
+//     LOG(INFO, "Deleting cached wizNetifConfigData");
+//     int result = SystemCache::instance().del(SystemCacheKey::WIZNET_CONFIG_DATA);
+//     return (result < 0) ? result : SYSTEM_ERROR_NONE;
+// }
+
+int WizNetifConfig::setConfigData(const WizNetifConfigData* configData) {
+    WizNetifConfigLock lk;
+
+    if (!initialized_) {
+        init();
+    } else {
+        validateWizNetifConfigData();
+    }
+
+    if (configData) {
+        memcpy(&wizNetifConfigData_, configData, sizeof(WizNetifConfigData));
+        // logWizNetifConfigData(wizNetifConfigData_);
+        saveWizNetifConfigData();
+    } else {
+        return SYSTEM_ERROR_BAD_DATA;
+    }
+
+    return SYSTEM_ERROR_NONE;
+}
+
+int WizNetifConfig::getConfigData(WizNetifConfigData* configData) {
+    WizNetifConfigLock lk;
+
+    if (!initialized_) {
+        init();
+    } else {
+        validateWizNetifConfigData();
+    }
+
+    if (configData) {
+        recallWizNetifConfigData();
+        memcpy(configData, &wizNetifConfigData_, sizeof(WizNetifConfigData));
+        // logWizNetifConfigData(wizNetifConfigData_);
+    }
+
+    return SYSTEM_ERROR_NONE;
+}
+
+int WizNetifConfig::validateWizNetifConfigData() {
+    if (wizNetifConfigData_.size != sizeof(WizNetifConfigData)) {
+        memset(&wizNetifConfigData_, 0, sizeof(wizNetifConfigData_));
+        wizNetifConfigData_.size = sizeof(WizNetifConfigData);
+        wizNetifConfigData_.cs_pin = PIN_INVALID;
+        wizNetifConfigData_.reset_pin = PIN_INVALID;
+        wizNetifConfigData_.int_pin = PIN_INVALID;
+
+        saveWizNetifConfigData();
```

Comment:
I would like to seperate this method to one that accepts the config data to be checked and validates it, and the other to set default config data and sync it to cache.